### PR TITLE
Add unit tests for etcd Client

### PR DESCRIPTION
**Pull request by jstordeur**

### CODE REVIEW

File: sd/etcd/client_test.go [hunk 1]
```diff
@@ -62,3 +65,249 @@ func TestOptions(t *testing.T) {
...
+	}
+}
+
+// getfakeClient return a new etcd.Client built on top of the mocked interfaces
+func getFakeClient(event, err chan bool, getres *getResult) (Client, error) {
+	return &client{
+		keysAPI: &fakeKeysAPI{event, err, getres},
+		ctx:     context.Background(),
+	}, nil
+}
```
Comment (peterbourgon) [inline sd/etcd/client_test.go]: No reason to have the return signature include an error that's always nil, unless I'm missing something?

```go
func newFakeClient(...) Client {
   return &client{...}
}
```

**Comment by peterbourgon** [comment]
Other than that one nit, looks good! For the record: this increases coverage of package etcd from 35.1% to 68.9%. [Here is the coverage report](http://peter.bourgon.org/etcd.html). Would you be up to adding some tests to registrar, too? ;)

File: sd/etcd/client_test.go (hunk 1 above)
Comment (jstordeur) [inline sd/etcd/client_test.go]: True, I'll change that

**Comment by jstordeur** [comment]
I'll have a look at registrar

**Comment by peterbourgon** [comment]
78.4% 😎

**Comment by peterbourgon** [comment]
Amazing! Thank you so much.

**Pull request merged**